### PR TITLE
Glue script ansible part

### PR DESCRIPTION
**Pull request by mpagot**

### CODE REVIEW

File: scripts/qesap/lib/config.py [hunk 1]
```diff
@@ -6,6 +6,26 @@
...
+    elif isinstance(value, dict):
+        param_value = ''
+        for dict_key, dict_value in value.items():
+            param_value = f'{param_value}\t{dict_key} = "{dict_value}"\n'
+        entry = f'{key} = {{\n' \
+                f'{param_value}' \
+                f'}}'
+    else:
+        log.error(f'Unrecognized value type in yaml file: {key} = {value}')
+        return False, ''
```
Comment (ricardobranco777) [inline scripts/qesap/lib/config.py:26]: Returning `entry` should be enough for a boolean test.

File: scripts/qesap/lib/config.py [hunk 2]
```diff
@@ -20,58 +40,76 @@ def yaml_to_tfvars(yaml_data):
...
+        configure_data (dict): configuration data structure
         tfvars_template (str): path to the tfvars template file
 
     Returns:
         bool: True(pass)/False(failure)
     """
+    log.info("Read %s", tfvars_template)
     with open(tfvars_template, 'r') as f:
-        tfvar_content = f.readlines()
+        tfvar_content = ['{}\n'.format(line.rstrip()) for line in f.readlines()]
```
Comment (ricardobranco777) [inline scripts/qesap/lib/config.py:92]: To pacify latest Pylint:

```suggestion
        tfvar_content = [f"{line.rstrip()}\n" for line in f.readlines()]
```

File: scripts/qesap/lib/config.py [hunk 3]
```diff
@@ -20,58 +40,76 @@ def yaml_to_tfvars(yaml_data):
...
 
+def terraform_yml(configure_data):
+    """
+    Check if Terraform:variables are present in the config.yaml
+    """
+    if not configure_data:
+        log.error("No configure_data")
+        return False
+
+    if 'terraform' not in configure_data.keys():
```
Comment (ricardobranco777) [inline scripts/qesap/lib/config.py:62]: `.keys()` here is redundant since Python 3.

File: scripts/qesap/lib/config.py [hunk 4]
```diff
@@ -20,58 +40,76 @@ def yaml_to_tfvars(yaml_data):
...
+        log.error("Missing 'terraform' key in configure_data")
+        return False
+    
+    if configure_data['terraform'] is None:
+        log.error("configure_data['terraform'] is empty")
+        return False
+
+    if 'variables' not in configure_data['terraform'].keys():
```
Comment (ricardobranco777) [inline scripts/qesap/lib/config.py:70]: ditto above.

File: scripts/qesap/test/conftest.py [hunk 5]
```diff
@@ -209,7 +265,8 @@ def check_multilines():
         """
     def _callback(lines):
         if len(lines) <= 1:
-            return False, "trento_cluster_install.sh should be a multi line script but it is only " + str(len(lines)) + " lines long."
+            return False,
+            "trento_cluster_install.sh should be a multi line script but it is only " + str(len(lines)) + " lines long."
```
Comment (ricardobranco777) [inline scripts/qesap/test/conftest.py:269]:
```suggestion
            f"trento_cluster_install.sh should be a multi line script but it is only {len(lines)} lines long."
```

**Review by ricardobranco777 — APPROVED** [review]
Some suggestions. Otherwise LGTM

**Review by lpalovsky — APPROVED** [review]
LGTM

**Review by alvarocarvajald — COMMENTED** [review]
Some comments and questions from my part.

File: scripts/qesap/lib/config.py [hunk 6]
```diff
@@ -6,72 +6,111 @@
...
+                if match:
+                    log.debug("Replace template %s with [%s = %s]", line, k, v)
+                    tfvar_content[index] = f"{k} = {v}\n"
+                    key_replace = True
+            # add the new key/value pair
+            if not key_replace:
+                log.debug("[k:%s = v:%s] is not in the template, append it", k, v)
+                entry = yaml_to_tfvars_entry(k, v)
+                if entry is None:
+                    return None
```
[the remark this answers is not in the document]
Comment (mpagot) [reply scripts/qesap/lib/config.py:113]: I like @lpalovsky opinion too about it, but I like to abruptly stop and return error if a key is misspelled or invalid in the config.yaml

File: scripts/qesap/lib/config.py (hunk 3 above)
Comment (mpagot) [reply scripts/qesap/lib/config.py:61]: I like this particular redundancy :-)

**Pull request merged**